Content.Shared/Stealth/SharedStealthSystem.cs

```diff
@@ -55,20 +59,32 @@ public virtual void SetEnabled(EntityUid uid, bool value, StealthComponent? comp
             return;
 
         component.Enabled = value;
-        Dirty(component);
+        Dirty(uid, component);
+    }
+
+    private void OnMobStateChanged(EntityUid uid, StealthComponent component, MobStateChangedEvent args)
+    {
+        if (args.NewMobState != MobState.Dead)
+        {
+            SetEnabled(uid, true, component);
+        }
+        else
+        {
+            SetEnabled(uid, false, component);
+        }
     }
```

Comment:
This should be a separate component, then probably raise an event when trying to set enabled and any callers can specify if stealth should actually be on or off.

Reply:
done